sandbox/modules/identity/src/main/java/org/opensearch/identity/realm/InternalRealm.java

```diff
@@ -139,16 +143,23 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
         throw new CredentialsException();
     }
 
-    public void setRealmName(String realmName) {
-        this.realmName = realmName;
+    @Override
+    protected AuthorizationInfo doGetAuthorizationInfo(PrincipalCollection principals) {
```

Comment:
What do we end up using this for? I am still not an expert on realms so not sure what the realm needs the permissions from the AuthorizationInfo here for. I know form the documentation that Shiro combines the results of getStringPermissions and getObjectPermissions but I don't know how this ties-in here. 

It grabs the authz info and then resolves the permissions for the user. But then the authorizations gets its ObjectPermissions set manually. Why do we want Shiro to set the Object Permissions after having just read the permissions?

Reply:
This resolves a subject to a list of permissions. So when `user1` logs in, this resolves the user's permissions.

Reply:
I see. Thank you for the explanation. That makes more sense. I just was not sure if we were somehow reading permissions and then reassigning them to the same object we just read them from.